app/src/titlebarwidget.cpp

```diff
@@ -0,0 +1,216 @@
+/*
+
+Pencil2D - Traditional Animation Software
+Copyright (C) 2005-2007 Patrick Corrieri & Pascal Naidon
+Copyright (C) 2012-2020 Matthew Chiawen Chang
+
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU General Public License
+as published by the Free Software Foundation; version 2 of the License.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+*/
+#include "titlebarwidget.h"
+
+#include <QHBoxLayout>
+#include <QVBoxLayout>
+#include <QLabel>
+#include <QIcon>
+#include <QToolButton>
+#include <QResizeEvent>
+#include <QStyle>
+#include <QPainter>
+#include <QMenu>
+#include <QDebug>
+
+#include "platformhandler.h"
+#include "buttonappearancewatcher.h"
+
+TitleBarWidget::TitleBarWidget(QWidget* parent)
+    : QWidget(parent)
+{
+
+    QVBoxLayout* vLayout = new QVBoxLayout();
+
+    vLayout->setContentsMargins(3,4,3,4);
+    vLayout->setSpacing(0);
+
+    mNormalTitleBarWidget = createNormalTitleBarWidget(this);
+
+    vLayout->addWidget(mNormalTitleBarWidget);
+
+    setLayout(vLayout);
+}
+
+TitleBarWidget::~TitleBarWidget()
+{
+}
+
+QWidget* TitleBarWidget::createNormalTitleBarWidget(QWidget* parent)
+{
+    bool isDarkmode = PlatformHandler::isDarkMode();
+    QWidget* containerWidget = new QWidget(parent);
+
+    mContainerLayout = new QHBoxLayout(parent);
+
+    mCloseButton = new QToolButton(parent);
+
+    mCloseButton->setStyleSheet(flatButtonStylesheet());
+
+    QSize iconSize = QSize(14,14);
+    QSize padding = QSize(2,2);
+
+    IconResource closeButtonRes;
+    closeButtonRes.lightMode = QIcon("://icons/themes/playful/window/window-close-button-normal.svg");
+    closeButtonRes.darkMode = QIcon("://icons/themes/playful/window/window-close-button-normal-darkm.svg");
+
+    QIcon closeIcon = closeButtonRes.iconForMode(isDarkmode);
+
+    IconResource closeHoverButtonRes;
+    closeHoverButtonRes.lightMode = QIcon("://icons/themes/playful/window/window-close-button-active.svg");
+    closeHoverButtonRes.darkMode = closeHoverButtonRes.lightMode;
+
+    mCloseButton->setIcon(closeIcon);
+    mCloseButton->setIconSize(iconSize);
+    mCloseButton->setFixedSize(iconSize + padding);
+    mCloseButton->installEventFilter(new ButtonAppearanceWatcher(closeButtonRes,
+                                                                 closeHoverButtonRes,
+                                                            this));
+
+    connect(mCloseButton, &QToolButton::clicked, this, [this] {
+        emit closeButtonPressed();
```

Comment:
Same here, the classic connection syntax should work?
```
    connect(mCloseButton, &QToolButton::clicked, this, &TitleBarWidget::closeButtonPressed);
```

Reply:
You're right, I think the lambda is from when i had some local logic involved.
Fixed.